### PR TITLE
Minor repl changes; minor conversion fixes

### PR DESCRIPTION
**Pull request by amosr**

### CODE REVIEW

File: main/icicle-repl.hs [hunk 1]
```diff
@@ -39,7 +39,7 @@ runRepl
     loop state
       = do line <- HL.getInputLine "> "
            case line of
-             Nothing      -> loop state
+             Nothing      -> return ()
```
Comment (tranma) [inline main/icicle-repl.hs]: with this, can `CommandBlank` (where user doesn't input anything) ever happen? if not I'll remove the case.

Comment (amosr) [inline main/icicle-repl.hs]: hmm, I think empty line will still do CommandBlank

Comment (amosr) [inline main/icicle-repl.hs]: the "Nothing" case is for EOF by the looks of it

Comment (tranma) [inline main/icicle-repl.hs]: cool, I'll leave CommandBlank in then.

see https://github.com/ambiata/icicle/pull/32

**Comment by tranma** [comment]
woah, we can have iREPL history too!!! let's?

**Comment by amosr** [comment]
iREPL history? I don't know what you mean

**Comment by tranma** [comment]
like ghci history, you can get commands from past sessions

**Comment by amosr** [comment]
if that's easy to do, it'd be super useful

File: src/Icicle/Source/ToCore/Base.hs [hunk 2]
```diff
@@ -91,6 +94,77 @@ freshly f
...
+   | anyPossibles
+   , Definitely <- universePossibility $ universe returns
+   = return $ mkSome $ X.makeApps f args
+   -- Otherwise, we have seen no possibles or the raw function
+   -- itself returns possibles, so we can leave the return as-is.
+   | otherwise
+   = return $          X.makeApps f args
+
+  -- For each entry of xts..
+  go ((x,u):xts') args anyPossibles
```
Comment (amosr) [inline src/Icicle/Source/ToCore/Base.hs]: am I a bad functional programmer because I wrote this with recursion instead of a "fold"?

Comment (amosr) [inline src/Icicle/Source/ToCore/Base.hs]: (I don't actually care)

Comment (tranma) [inline src/Icicle/Source/ToCore/Base.hs]: a fundamentalist functional programmer!

**Comment by amosr** [comment]
gonna merge this. there's some cool stuff here: converting from Icicle Source to Core mainly

**Pull request merged**